lib/deimos/instrumentation.rb

```diff
@@ -48,11 +48,17 @@ def self.send_produce_error(event)
       messages.group_by(&:topic).each do |topic, batch|
         next if batch.empty?
 
-        producer = batch.first.metadata[:producer_name]
-        payloads = batch.map { |m| m.metadata[:decoded_payload] }
+        producer = Deimos::Producer.descendants.find { |c| c.topic == topic }
+        decoder = Deimos::KafkaMessage.decoder(topic)
+        payloads = batch.map do |m|
+          decoder&.decoder&.decode(m.value)
+        end
```

Comment:
I think this is a bit convoluted. There's an easier way to do this now:
```ruby
decoder = Deimos.schema_backend(schema: producer.config[:schema],
                                           namespace: producer.config[:namespace])
payloads = batch.map { |m| decoder.decode(m.value) }
```
Also don't forget to handle the case where the producer wasn't found.